vm/src/vm.rs

```diff
@@ -266,14 +266,10 @@ impl VirtualMachine {
                 ref function,
                 ref object,
             } => self.invoke(function.clone(), args.insert(object.clone())),
-            PyObjectPayload::Instance { .. } => {
-                self.call_method_pyargs(&func_ref, "__call__", args)
-            }
-            ref payload => {
-                // TODO: is it safe to just invoke __call__ otherwise?
-                trace!("invoke __call__ for: {:?}", payload);
+            PyObjectPayload::Instance { .. } | PyObjectPayload::WeakRef { .. }  => {
```

Comment:
Why the more strict check here? This looks like it will break callable subtypes of ```int```, for example.

Reply:
Oh, right, that's true. I was expecting Python `int` subtypes to be Payload::Instance and was wrong about it. Will fix in a moment.

Reply:
This is one of the flaws in our object model, btw. A subclass of a builtin type should have a  non-Instance Payload and an Instance dictionary, we only support having one or the other.